asapdiscovery-data/asapdiscovery/data/schema_v2/ligand.py

```diff
@@ -0,0 +1,151 @@
+from pathlib import Path
+from typing import Any, Dict, Optional, Tuple, Union  # noqa: F401
+from uuid import UUID
+
+from asapdiscovery.data.openeye import (
+    get_SD_data,
+    get_SD_data_dict,
+    oechem,
+    oemol_to_inchi,
+    oemol_to_inchikey,
+    oemol_to_sdf_string,
+    oemol_to_smiles,
+    print_SD_Data,
+    sdf_string_to_oemol,
+    set_SD_data,
+    set_SD_data_dict,
+    smiles_to_oemol,
+)
+from asapdiscovery.data.schema import ExperimentalCompoundData
+from pydantic import UUID4, Field
+
+from .schema_base import (
+    DataModelAbstractBase,
+    DataStorageType,
+    read_file_directly,
+    write_file_directly,
+)
+
+
+class InvalidLigandError(ValueError):
+    ...
+
+
+# Ligand Schema
+
+
+class LigandIdentifiers(DataModelAbstractBase):
+    """
+    Identifiers for a Ligand
+    """
+
+    moonshot_compound_id: Optional[str] = Field(
+        None, description="Moonshot compound ID"
+    )
+    manifold_api_id: Optional[UUID] = Field(
+        None, description="Unique ID from Postera Manifold API"
+    )
+    manifold_vc_id: Optional[str] = Field(
+        None, description="Unique VC ID (virtual compound ID) from Postera Manifold"
+    )
+    compchem_id: Optional[UUID4] = Field(
+        None, description="Unique ID for P5 compchem reference"
+    )
+
+
+class Ligand(DataModelAbstractBase):
+    """
+    Schema for a Ligand
+    """
+
+    compound_name: str = Field(None, description="Name of compound")
+    ids: Optional[LigandIdentifiers] = Field(
+        None,
+        description="LigandIdentifiers Schema for identifiers associated with this ligand",
+    )
```

Comment:
I think we would want to make at least 1 ID a requirement?

Reply:
Seems reasonable, I will move `compound_name` to `LigandIdentifiers` for this purpose.

Reply:
we decided to not move this, instead we'll have a validator check to make sure that at least compound_name or one of the ids is set

Reply:
Done!